server/connectors/api-connector-jdbc/src/main/scala/com/prisma/api/connector/jdbc/impl/DeleteNodeInterpreters.scala

```diff
@@ -94,38 +94,45 @@ trait CascadingDeleteSharedStuff {
   implicit def ec: ExecutionContext
 
   def performCascadingDelete(mutationBuilder: JdbcActionsBuilder, model: Model, parentId: IdGCValue): DBIO[Unit] = {
-    val actions = model.cascadingRelationFields.map { field =>
-      recurse(
-        mutationBuilder = mutationBuilder,
-        parentField = field,
-        parentIds = Vector(parentId),
-        visitedModels = Vector(model)
-      )
-    }
+    val actions = model.cascadingRelationFields.map(field => recurse(mutationBuilder = mutationBuilder, parentField = field, parentIds = Vector(parentId)))
     DBIO.seq(actions: _*)
   }
 
   private def recurse(
       mutationBuilder: JdbcActionsBuilder,
       parentField: RelationField,
       parentIds: Vector[IdGCValue],
-      visitedModels: Vector[Model]
+      seenIds: Vector[IdGCValue] = Vector.empty
   ): DBIO[Unit] = {
 
     for {
       ids        <- mutationBuilder.getNodeIdsByParentIds(parentField, parentIds)
       groupedIds = ids.grouped(10000).toVector
       model      = parentField.relatedModel_!
-      _          = if (visitedModels.contains(model)) throw APIErrors.CascadingDeletePathLoops()
       //nestedActions
-      nextCascadings = model.cascadingRelationFields.filter(_ != parentField)
-      childActions = for {
-        field   <- nextCascadings
-        idGroup <- groupedIds
-      } yield {
-        recurse(mutationBuilder, field, idGroup, visitedModels :+ model)
-      }
-      _ <- DBIO.seq(childActions: _*)
+      _ <- if (ids.isEmpty) {
+            DBIO.successful(())
+          } else {
+            //children
+            val cascadingChildren = model.cascadingRelationFields.filter(_ != parentField.relatedField)
+            val childActions = for {
+              field   <- cascadingChildren
+              idGroup <- groupedIds
+            } yield {
+              recurse(mutationBuilder, field, idGroup)
+            }
+            //other parent
+            val cascadingParent = model.cascadingRelationFields.filter(_ == parentField.relatedField)
```

Comment:
This can always just find one field, right? If this is true, a call to `find` or `headOption` after the filter would make that more obvious.

Reply:
Also the name of the variable is a bit misleading maybe. This is a field on the child model and points to the parent. Right now it reads as if it is a parent field actually.